app/jobs/doi_verification_job.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class DoiVerificationJob
+  def perform(publication)
+    if publication.doi.present?
+      DoiVerificationService.new(publication).verify
+    else
+      response = UnpaywallClient.new.query_unpaywall(publication)
+      if publication.title&.downcase&.gsub(/[^a-z0-9]/, '') == response.title&.downcase&.gsub(/[^a-z0-9]/, '') && response.doi.present?
```

Comment:
I've since noticed some of this same logic in other places. For example [here](https://github.com/psu-libraries/researcher-metadata/blob/1891128e167193b8a3b485e863f222a8d3aa7ee3/app/models/publication_merge_on_matching_policy.rb#L98) and [here](https://github.com/psu-libraries/researcher-metadata/blob/1891128e167193b8a3b485e863f222a8d3aa7ee3/app/models/publication_merge_on_matching_policy.rb#L103).

Maybe it makes sense for `Publication` and `UnpaywallResponse` to each have a method like `#matchable_title` or something of that nature..

Reply:
Would it make more sense to add `#matchable_title` to `Publication` and `UnpaywallResponse` and have it just format that class's title for easy comparison or to add `#title_match?` to `Publication`, have it take in a string, handle all matching logic, and return true/false (similar to [here](https://github.com/psu-libraries/researcher-metadata/blob/1891128e167193b8a3b485e863f222a8d3aa7ee3/app/importers/unpaywall_publication_importer.rb#L137))?

Reply:
Good question - I'm not sure. Probably the latter if it serves all of the existing use cases.